packages/angular-query-experimental/src/__tests__/inject-mutation.test.ts

```diff
@@ -455,4 +455,38 @@ describe('injectMutation', () => {
 
     await expect(() => mutateAsync()).rejects.toThrowError(err)
   })
+
+  test('should execute callback in injection context', async () => {
+    @Injectable()
+    class FakeService {
+      updateData(name: string) {
+        return Promise.resolve(name)
+      }
+    }
+
+    @Component({
+      selector: 'app-fake',
+      template: ``,
+      standalone: true,
+      providers: [FakeService],
+    })
+    class FakeComponent {
+      mutation = injectMutation(() => {
+        const service = inject(FakeService)
+        return {
+          mutationFn: (name: string) => service.updateData(name),
+        }
+      })
+    }
+
+    const fixture = TestBed.createComponent(FakeComponent)
+    fixture.detectChanges()
+
+    // check if injection contexts persist in a different task
+    await new Promise<void>((resolve) => queueMicrotask(() => resolve()))
```

Comment:
Without this, the test will pass with the previous implementation. That is because the callback of `effect` which causes the exception will run in a different task.